### ISSUE TITLE
Private registry is not applied for the first rancher-agent container during Infrastructure Provider cluster creation

### ISSUE THREAD

**Issue by bashofmann** [rancher/rancher#26916]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
* Create a cluster on an Infrastructure Provider with a Node Template
* Configure a private registry
* SSH into a node and observer the Docker images that are being started there, e.g. with `docker ps`

**Result:**
The image of the first `rancher/rancher-agent` container is pulled from Docker Hub and not from the provided private registry. All other images are pulled from the private registry afterwards.

Example output with Rancher 2.3.6:

```
sudo docker ps
CONTAINER ID        IMAGE                                                               COMMAND                  CREATED              STATUS              PORTS               NAMES
e9ada19cdfc9        nexus-registry.rancher-aws.plgrnd.be/rancher/rancher-agent:v2.3.6   "run.sh --server htt…"   About a minute ago   Up About a minute                       friendly_ishizaka
1e2e7bc095f6        nexus-registry.rancher-aws.plgrnd.be/rancher/rancher-agent:v2.3.6   "run.sh -- share-roo…"   About a minute ago   Up About a minute                       share-mnt
5ff0bd62ec6b        rancher/rancher-agent:v2.3.6                                        "run.sh --server htt…"   2 minutes ago        Up 2 minutes                            pensive_elbakyan
```

The issue also is still there with 2.4.2.

**Other details that may be helpful:**
* I only tested this with the EC2 provider so far, if needed I can also test it with other providers.
* When creating a custom cluster, the printed `docker run` command correctly uses the image from the private registry.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.4.2
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Infrastructure Provider
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud, ec2 t2.large
- Kubernetes version (use `kubectl version`): 1.17.4-rancher1-3
- Docker version (use `docker version`): 19.03

**Comment by superseb** [comment]
This should be fixed via https://github.com/rancher/rancher/issues/26366

**Comment by deniseschannon** [comment]
This appears to have already been fixed as of v2.4.13, which means it also would be fixed as of 2.5.6+
